tests/e2e/test_rosacli_policy.go

```diff
@@ -388,20 +388,13 @@ var _ = Describe("Validation testing",
 				Expect(err).To(BeNil())
 				_, err = awsClient.CreateRegularRole(notRHManagedRoleName, notRHManagedRolePolicy)
 				Expect(err).To(BeNil())
-				testingRolesToClean = append(testingRolesToClean, notRHManagedRoleName)
-
-				By("Prepare 10 arbitrary policies for testing")
-				awsClient, err = aws_client.CreateAWSClient("", "")
-				Expect(err).To(BeNil())
-
-				for i := 0; i < 10; i++ {
-					arn, err := awsClient.CreatePolicy(
-						fmt.Sprintf("ocmqe-arpolicy-%s-%d", helper.GenerateRandomString(3), i),
-						statement,
-					)
+				defer func() {
+					By("Detach the attached policy " + notRHManagedRolePolicy + " from role" + notRHManagedRoleName)
+					err := awsClient.DetachIAMPolicy(notRHManagedRoleName, notRHManagedRolePolicy)
```

Comment:
OCP-74225 has fixed by https://github.com/openshift/rosa/pull/2492 
The fix swaps places of "By("Delete arbitrary policies")" and "By("Delete the testing role")" to avoid the failure when delete arbitrary policies.